### PR TITLE
Update sharp to 0.33 to fix issue with Alpine Linux

### PR DESCRIPTION
**Pull request by PeterDraex**
## Changes

This PR updates the version of the `sharp` dependency to `^0.33` and thus makes Astro compatible with Alpine Linux containers.

It was impossible for me to use the Astro image service in an Alpine Linux docker container. Even though I would install sharp `0.33.3` in my app, pnpm would download version `0.32.6`, which doesn't work in Alpine Linux container. I have to override the downloaded version in my package.json. 

For more details about the issue see:
- https://discord.com/channels/830184174198718474/1224861729792458803/1224861729792458803
- https://github.com/lovell/sharp/issues/4054

## Testing

- Automatic tests
- Override of sharp with the current version in local project

## Docs

This resolves existing bug. No documentation needed.

### CODE REVIEW

**Review by Princesseuh — CHANGES_REQUESTED** [review]
Blocking this because it's a breaking change. Sharp's 0.33.0 `engines` contract is stricter than Astro's, requiring a higher version of Node 18 and 20 than Astro, and not supporting Node 19 at all. 

As such, for a certain percentage of users, Sharp would just not get installed ever unless they forced the install. Last time we updated to 0.33.0 (because I didn't know its checks were stricter than us, ha), we ended up with a bunch of support threads from people running outdated versions of Node.

File: packages/astro/package.json [hunk 1]
```diff
@@ -179,7 +179,7 @@
     "zod-to-json-schema": "^3.22.4"
   },
   "optionalDependencies": {
-    "sharp": "^0.32.6"
+    "sharp": "^0.33"
```
Comment (Princesseuh) [inline packages/astro/package.json:182]:
```suggestion
    "sharp": "^0.33.3"
```

**Review by Princesseuh — APPROVED** [review]
Thank you, this looks great!

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -694,8 +694,8 @@ importers:
         version: 3.22.5(zod@3.22.4)
     optionalDependencies:
       sharp:
-        specifier: ^0.32.6
-        version: 0.32.6
+        specifier: ^0.33
```
Comment (Princesseuh) [inline pnpm-lock.yaml:697]:
```suggestion
        specifier: ^0.33.3
```

**Review by ematipico — APPROVED** [review]

File: .changeset/famous-mugs-grab.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+Update sharp to 0.33 to fix issue with Alpine Linux
```
Comment (sarah11918) [inline .changeset/famous-mugs-grab.md:5]:
```suggestion
Updates sharp to version 0.33 to fix issues when using an Alpine Linux container.
```

**Review by sarah11918 — COMMENTED** [review]
Thanks for this helpful contribution @PeterDraex !  I've just updated the changeset message to be a tiny bit more explicit. Especially since this is being released with a minor version of Astro, we try to give the changesets a little extra love and attention as people are checking more closely!  :raised_hands:

**Comment by PeterDraex** [comment]
> @PeterDraex would you mind rebasing with `main` and rerunning install to update the lockfile?

@matthewp Done. I've also updated sharp version in `examples/starlog/package.json`. Is that okay?

**Pull request merged**